### PR TITLE
Tools: Cleanup and update other bash tools

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

* Add color to tool output!
* Update check-development-environment.sh with checks for jq and modern bash.
* Make tools/revert-release.sh generic.

#### Jetpack product discussion

None.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Do things work?

#### Proposed changelog entry for your changes:

* None needed.

### CODE REVIEW

File: tools/includes/proceed_p.sh [hunk 1]
```diff
@@ -14,16 +16,22 @@ fi
...
 	# Clear input before prompting
 	while read -r -t 0 OK; do read -r OK; done
 
-	[[ "$1" ]] && M="$1 " || M=""
-	read -n 1 -p "$M${2:-Proceed?} [y/N] " OK
+	local PROMPT
+	[[ -n "$1" ]] && PROMPT="$1 "
+	PROMPT="${PROMPT}${2:-Proceed?} [y/N] "
+	if color_supported; then
+		PROMPT=$(FORCE_COLOR=1 info "$PROMPT")
```
Comment (kraftbj) [inline tools/includes/proceed_p.sh:31]:
```suggestion
		PROMPT=$(FORCE_COLOR=1 prompt "$PROMPT")
```

Set a different color to stand out for a question.

File: tools/includes/chalk-lite.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+function error {
+	chalk '1;31' "$@" >&2
+}
+
+function die {
+	error "$@"
+	exit 1
+}
+
```
Comment (kraftbj) [inline tools/includes/chalk-lite.sh:81]:
```suggestion

function prompt {
	yellow "$@"
}

```

**Review by kraftbj — CHANGES_REQUESTED** [review]
I really like this! Only suggestion is to have a more stand-out color for a prompt with a suggestion of yellow.

File: tools/includes/proceed_p.sh (hunk 1 above)
Comment (anomiex) [reply tools/includes/proceed_p.sh:31]: Sure, no objection.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**